src/SonarLint.VSSpecificAssemblies.props

```diff
@@ -157,29 +157,29 @@
   <!-- ******************************************************************************* -->
   <!-- Microsoft.VisualStudio.Shell assembly -->
   <!-- ******************************************************************************* -->
-  <ItemGroup Label="Shell versioned references" Condition="$(RequiresVersionSpecificShell)=='true' AND '$(VsTargetVersion)'=='2015'">
-    <PackageReference Include="Microsoft.VisualStudio.Shell.14.0"
-                      Version="14.3.25407"
-                      GeneratePathProperty="true"
-                      ExcludeAssets="all"
-                      PrivateAssets="all">
-    </PackageReference>
-    <ReferencePath Include="$(NuGetPackageRoot)\Microsoft.VisualStudio.Shell.14.0\14.3.25407\lib\Microsoft.VisualStudio.Shell.14.0.dll">
-      <Private>$(CopyVSAssembliesToOutput)</Private>
-      <Aliases>global, versionSpecificShell</Aliases>
-    </ReferencePath>
+  <PropertyGroup>
+    <RequiresShell14 Condition="$(RequiresVersionSpecificShell)=='true' AND '$(VersionSpecificSuffix)'=='2015'">true</RequiresShell14>
+    <RequiresShell15 Condition="$(RequiresVersionSpecificShell)=='true' AND ('$(VersionSpecificSuffix)'=='2017' OR '$(VersionSpecificSuffix)'=='2019')">true</RequiresShell15>
+  </PropertyGroup>
+
+  <ItemGroup Label="Shell 14 references" Condition="$(RequiresShell14)=='true'">
+    <PackageReference Include="Microsoft.VisualStudio.Shell.14.0" Version="14.3.25407"/>
   </ItemGroup>
-  <ItemGroup Label="Shell versioned references" Condition="$(RequiresVersionSpecificShell)=='true' AND '$(VsTargetVersion)'!='2015'">
-    <PackageReference Include="Microsoft.VisualStudio.Shell.15.0"
-                      Version="15.0.26201"
-                      GeneratePathProperty="true"
-                      ExcludeAssets="all"
-                      PrivateAssets="all">
-    </PackageReference>
-    <ReferencePath Include="$(NuGetPackageRoot)\Microsoft.VisualStudio.Shell.15.0\15.0.26201\lib\Microsoft.VisualStudio.Shell.15.0.dll">
-      <Private>$(CopyVSAssembliesToOutput)</Private>
-      <Aliases>versionSpecificShell</Aliases>
-    </ReferencePath>
+  <ItemGroup Label="Shell 15 references" Condition="$(RequiresShell15)=='true'">
+    <PackageReference Include="Microsoft.VisualStudio.Shell.15.0" Version="15.0.26201"/>
   </ItemGroup>
+  <Target Name="SilenceShell14" Condition="$(RequiresShell15)=='true'" BeforeTargets="FindReferenceAssembliesForReferences;ResolveReferences">
```

Comment:
SilenceShell14 will be used by IssueViz.Security and IssueViz.Security.Tests - it's to avoid conflicts caused by shell 14 coming from IssueViz assembly.

Reply:
It's probably adding a comment at the top of this section explaining the overall scenario e.g.

>Projects won't build if they reference both Shell 14 and Shell 15. Each project will only directly reference one Shell version. However, it could transitively reference the other Shell assembly e.g. X depends on Shell14 and Y, Y depends on Shell15. In this case, we need to remove the transitive references to Shell15 when building X.